### ISSUE TITLE
Non-tools in the "hand" inventory list cause a segfault

### ISSUE THREAD

**Issue by AlexYst** [minetest/minetest#7215]
##### Issue type
 - Bug report
 
##### Minetest version
```
0.4.16
```

##### OS / Hardware
Operating system: Debian 9
CPU: Irrelevant

##### Summary
If an item without tool capabilities is put in the "hand" inventory list of a player, a segfault is caused.

##### Steps to reproduce
Simply put something such as cobble in the "hand" inventory list. For example, you could use this code and start a new player:

```
minetest.register_on_newplayer(function(player)
	local inv = player:get_inventory()
	inv:add_item("hand", "default:cobble")
end)
```

There may be no good reason to put cobble into this list, but it shouldn't crash the game to try. Such items could be rejected by the inventory if need be. The segfault happens only once the hotbar slot selected contains no item.

**Comment by rubenwardy** [comment]
The right behaviour here is to treat it as a null capability - disallow digging or whatever with the hand, only tools with their own capabilities should work

**Comment by AlexYst** [comment]
Adding a null, default capacity isn't too hard. Just adding the table for such null capacity to the default item definition tables seems to completely fix the segfault, actually: <https://github.com/minetest/minetest/pull/7216>.

**Comment by TeTpaAka** [comment]
I tested it and https://github.com/minetest/minetest/commit/7786521f151b6006656768fdd308f1d7e1635645 seems to have fixed the issue in current master.

**Comment by SmallJoker** [comment]
Cannot reproduce this in 0.4.17-dev. Seems to be fixed since the commit pointed out above (#5899).
